pkg/api/server/v1alpha2/ordering.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 The Tekton Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func orderBy(fields string) (string, error) {
+	if strings.TrimSpace(fields) == "" {
+		return "", nil
+	}
+
+	var orderBy []string
+	for _, field := range strings.Split(fields, ",") {
+		field = strings.TrimSpace(field)
+		f := strings.Split(field, " ")
+		switch len(f) {
+		case 1:
+			orderBy = append(orderBy, strings.TrimSpace(f[0]))
```

Comment:
I think we're going to need a bit more logic here, since the database schema doesn't match the API schema:

- the `name` field in the API for results actually maps to (`parent`, `name`) in the database schema, so names can potentially be out of order depending on their parent. 
- Timestamps have the reverse problem- the API timestamp has the subfields (`seconds`, `nanos`), but the timestamp in the schema is closer to a `time.Time`.

We'll also want some documentation and validation to give clear errors on what fields are supported - I don't think we have the ability to order by fields in the data blob.

Reply:
Ah ok, thanks for this.  I need to dig around and understand what's going on a bit more thoroughly.

Reply:
I've limited the fields which can be ordered to just `created_time` and `updated_time` now.  So this limits the field types we're sorting on to just timestamps in this PR.

I'm curious about the API timestamp vs schema timestamp issue you mention - I'm not totally clear where that comes in to play yet.

I've added a short documentation snippet to `docs/api/README.md` describing the order_by parameter.